R/translate-sql-odbc.R

```diff
@@ -35,7 +50,79 @@ base_odbc_agg <- sql_translator(
 base_odbc_win <- sql_translator(
   .parent = base_win,
   n             = function() sql("COUNT(*)"),
-  count         = function() sql("COUNT(*)")
+  count         = function() sql("COUNT(*)"),
+
+  # Window functions in this variant use field that its being
+  # 'windowed' as the secondary ORDER BY instead of the field
+  # from win_current_order().  In the tests, having an ORDER BY
```

Comment:
Yes sir, 

I ran this code on two machines, one with the code from this PR and another with CRAN versions:

```
m <- tbl(con, "mtcars") %>%
  mutate(next_mpg = lead(mpg))

show_query(m)

m
```

CRAN version

```
<SQL>
SELECT "row_names", "mpg", "cyl", "disp", "hp", 
"drat", "wt", "qsec", "vs", "am", "gear", "carb", LEAD("mpg") AS "next_mpg"
FROM "mtcars"
 Show Traceback
Error in new_result(connection@ptr, statement) : nanodbc/nanodbc.cpp:1344: HY000: The function 'LEAD' must have an OVER clause., Statement(s) could not be prepared.,
```
From this PR, the query is successful:

```
<SQL>
SELECT "row_names", "mpg", "cyl", "disp", "hp", 
"drat", "wt", "qsec", "vs", "am", "gear", "carb", LEAD("mpg", 1) OVER (ORDER BY "mpg") AS "next_mpg"
FROM "mtcars"
```

Here is one using `group_by` to activate `PARTITION`: 

```
m <- tbl(con, "mtcars") %>%
  group_by(am) %>%
  mutate(next_mpg = lead(mpg))

show_query(m)

m
```
From CRAN:

```
<SQL>
SELECT "row_names", "mpg", "cyl", "disp", "hp",
 "drat", "wt", "qsec", "vs", "am", "gear", "carb", LEAD("mpg") AS "next_mpg"
FROM "mtcars"
 Show Traceback
Error in new_result(connection@ptr, statement) : nanodbc/nanodbc.cpp:1344: HY000: The function 'LEAD' must have an OVER clause., Statement(s) could not be prepared.,
```
From this PR, the query is successful:

```
<SQL>
SELECT "row_names", "mpg", "cyl", "disp", "hp", 
"drat", "wt", "qsec", "vs", "am", "gear", "carb", LEAD("mpg", 1) OVER (PARTITION BY "am" ORDER BY "mpg") AS "next_mpg"
FROM "mtcars"
```